ra_aid/file_listing.py

```
@@ -0,0 +1,131 @@
+        DirectoryNotFoundError: If directory does not exist
+        DirectoryAccessError: If directory cannot be accessed
+        GitCommandError: If git command fails unexpectedly
+    """
+    try:
+        path = Path(directory).resolve()
+        if not path.exists():
+            raise DirectoryNotFoundError(f"Directory does not exist: {directory}")
+        if not path.is_dir():
+            raise DirectoryNotFoundError(f"Path is not a directory: {directory}")
+
+        result = subprocess.run(
```

Ruff (0.8.2)

`subprocess` call: check for execution of untrusted input

(S603)